plugin-server/src/worker/ingestion/event-pipeline/embedErrorEventStep.ts

```diff
@@ -0,0 +1,92 @@
+import { FeatureExtractionPipeline, FeatureExtractionPipelineOptions } from '@xenova/transformers'
+
+import { defaultConfig } from '../../../config/config'
+import { runInstrumentedFunction } from '../../../main/utils'
+import { PreIngestionEvent } from '../../../types'
+import { status } from '../../../utils/status'
+import { EventPipelineRunner } from './runner'
+
+const errorEmbeddingModel = defaultConfig.ERROR_EVENT_EMBEDDING_MODEL
+const errorEmbeddingEnabledTeams = (defaultConfig.ENABLE_ERROR_EMBEDDING_TEAM_IDS || '')
+    .split(',')
+    .map((id) => parseInt(id, 10))
+
+let featureExtractionPipeline: FeatureExtractionPipeline | null
+
+// this downloads the model the first time it is run, so eventually we'll do this on server start-up
+// for now we only want to spend the cost of loading the model if we are actually using it
+export async function initEmbeddingModel(): Promise<FeatureExtractionPipeline | null> {
```

Comment:
We want to do all the heavy work before we connect the kafka consumers, to avoid blocking after we're assigned a partition. I think it's better to call it at init (before we init the consumers), and fast exit if `ENABLE_ERROR_EMBEDDING_MAX_TEAM_ID === 0`.

Let's make sure that this condition (or a more explicit boolean config field) disables all these bits in case we're causing an incident.

Reply:
ok, so the library will load from local file or from a configured remote host (instead of hugging face).

i guess i can mirror the structure per model in S3 with cloud front in front of it 

so something like this in the code so we only support this when the remote host is configured

```
const errorEmbeddingModel = defaultConfig.ERROR_EVENT_EMBEDDING_MODEL
if (defaultConfig.ERROR_EMBEDDING_REMOTE_HOST === null) {
    // this effectively disables this feature
    env.allowRemoteModels = false
    env.allowLocalModels = false
} else {
    env.remoteHost = defaultConfig.ERROR_EMBEDDING_REMOTE_HOST
}
```

@frankh I'm assuming S3 with cloudfront is the easiest way? I can do terrible terraform to upset you in the morning 🙈